### ISSUE TITLE
Vert.x 4 Serialization error (Native mode)

### ISSUE THREAD

**Issue by pjgg** [quarkusio/quarkus#16555]
## Describe the bug

Quarkus: 2.X Native mode.

I am working in a Vert.x application over Quarkus 2.x and I am getting the following error when I am trying to encode a `io.vertx.ext.auth.User` into a Json String through this method `Json.encode(user)` (Note that the user is empty). 

Error: 

```
Failed to encode as JSON: No serializer found for class io.vertx.ext.auth.impl.UserImpl and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)"
```

I have tried to register an auth "user", for reflection through a `reflection-config.json` file or by a `@RegisterForReflection(targets = {User.class, UserImpl.class})` annotation without success. 

For example:

```
package io.quarkus.qe.vertx.web.auth;

import io.quarkus.runtime.annotations.RegisterForReflection;
import io.vertx.ext.auth.User;
import io.vertx.ext.auth.impl.UserImpl;

@RegisterForReflection(targets = {User.class, UserImpl.class})
public class MyReflectionConfiguration {
}
```

In the end following the error recommendation we have created a custom object mapper used by Vertx `Json.encode` library. 

```
@Singleton
public class VertxAuthObjectMapperCustomizer implements ObjectMapperCustomizer {

    @Override
    public void customize(ObjectMapper objectMapper) {
        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
    }
}

```

You have a reproducer in [this project](https://github.com/quarkus-qe/beefy-scenarios)  over [this module](https://github.com/quarkus-qe/beefy-scenarios/tree/main/302-quarkus-vertx-jwt). You could clone the project, remove the class `io.quarkus.qe.vertx.web.config.VertxAuthObjectMapperCustomizer` and run the following command: `mvn clean install -pl 302-quarkus-vertx-jwt -P native` 

Another way to reproduce the issue is just to create a simple Vertx app for example with Reactive-routes:

Dependencies: 
```
 <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-vertx-web</artifactId>
        </dependency>

        <dependency>
            <groupId>io.vertx</groupId>
            <artifactId>vertx-auth-jwt</artifactId>
        </dependency>

```

And then create the following endpoint:
```
@RouteBase(path = "/test")
public class TestRouteHandler {
   
    @Route(methods = HttpMethod.GET, path = "/hello")
    Uni<String> hello() {
        User user = new UserImpl();
        System.out.println(Json.encode(user));
        return Uni.createFrom().item("hello");
    }
}
```

Make a native compilation and then make a request to `http://localhost:8080/test/hello/`

Or even forget about `io.vertx:vertx-auth-jwt`

IF you have a "RandomObject" like 

```
@RegisterForReflection
public class RandomObject {
}
```

And then you try to serialize as a Json, is going to fail (native mode):

```
@RouteBase(path = "/test")
public class TestRouteHandler {

    @Route(methods = HttpMethod.GET, path = "/hello")
    Uni<String> hello() {
        RandomObject random = new RandomObject();
        System.out.println(Json.encode(random));
        return Uni.createFrom().item("hello");
    }
}
```

Query: `curl -v http://localhost:8080/test/hello/`
Result: 
```
 io.vertx.core.json.EncodeException: Failed to encode as JSON: No serializer found for class io.quarkus.qe.RandomObject and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)
````

**Comment by cescoffier** [comment]
Yes, this case cannot be easily tracked.
Note that it should work if you return a `Uni<RandomObject>` directly.

**Comment by cescoffier** [comment]
BTW, was it working with the 1.x version?

**Comment by NinjaTycoon** [comment]
I had an issue with a 3rd party library I created that uses gson.  To my surprise, gson itself basically works without a problem in the native.  but, it was only when I saw this syntax above that I discovered how to annotate 3rd party library classes inside the main project:

@RegisterForReflection(targets = {User.class, UserImpl.class})

Sort of off topic, but just wanted a quick way to suggest that the documentation have this added in various locations as I read a ton of documentation yet never saw this usage explained.  Only here in this Issue did I discover it. Saved the day!!!  Thanks!
